Content.Server/GameTicking/Rules/RevolutionaryRuleSystem.cs

```diff
@@ -0,0 +1,392 @@
+                }
+                if (_mindSystem.TryGetSession(mindId, out var session) && comp != null)
+                {
+                    comp.HeadRevs.Add(inCharacterName, mindId);
+                }
+            }
+            _antagSelectionSystem.GiveAntagBagGear(headRev, "RevFlash");
+            _antagSelectionSystem.GiveAntagBagGear(headRev, "ClothingEyesGlassesSunglasses");
+            EnsureComp<RevolutionaryComponent>(headRev);
+            EnsureComp<HeadRevolutionaryComponent>(headRev);
+        }
+    }
+
+    /// <summary>
+    /// At start gets all the initial players to check for heads
+    /// </summary>
+    private void InitialAssignCommandStaff()
+    {
+        var playerList = new List<EntityUid>();
+        var query = AllEntityQuery<MindContainerComponent>();
+        while (query.MoveNext(out var uid, out var comp))
+        {
+            if (HasComp<HumanoidAppearanceComponent>(uid))
+            {
+                playerList.Add(uid);
+            }
+        }
+        AssignCommandStaff(playerList);
+    }
+
+    /// <summary>
+    /// Gives heads the command component so they are tracked for Revs.
+    /// </summary>
+    /// <param name="playerList"></param>
+    private void AssignCommandStaff(List<EntityUid> playerList, bool latejoin = false)
+    {
+        var currentCommandStaff = 0;
+        var jobs = _prototypeManager.Index<DepartmentPrototype>("Command");
+        foreach (var player in playerList)
+        {
+            if (_mindSystem.TryGetMind(player, out var mindId, out var mind))
+            {
+                if (_jobSystem.MindTryGetJob(mindId, out var jobComp, out var jobProto) && jobs.Roles.Contains(jobProto.ID) && mind.OriginalOwnedEntity != null)
+                {
+                    if (!HasComp<HeadRevolutionaryComponent>(mind.OriginalOwnedEntity))
+                    {
+                        EnsureComp<CommandStaffComponent>(mind.OriginalOwnedEntity.Value);
+                        currentCommandStaff++;
+                    }
+                }
+            }
+        }
+        if (latejoin == false)
+        {
+            if (currentCommandStaff == 0)
+            {
+                var query = AllEntityQuery<RevolutionaryRuleComponent, GameRuleComponent>();
+                while (query.MoveNext(out var uid, out var comp, out var gameRule))
+                {
+                    GameTicker.EndGameRule(uid, gameRule);
+                }
+            }
+        }
+    }
+
+    /// <summary>
+    /// Called when a Head Rev uses a flash in melee to convert somebody else.
+    /// </summary>
+    public void OnPostFlash(EntityUid uid, HeadRevolutionaryComponent comp, ref AfterFlashedEvent ev)
+    {
+        var stunTime = TimeSpan.FromSeconds(3);
+        if (!HasComp<RevolutionaryComponent>(ev.Target) && !HasComp<MindShieldComponent>(ev.Target) && (HasComp<HumanoidAppearanceComponent>(ev.Target) || HasComp<MonkeyAccentComponent>(ev.Target))
```

Comment:
dont hardcode checking for monkeyaccent, if the intent here is to allow monkeys to be converted just add a new component called like `AlwaysRevolutionaryConvertible` or something and give it to monkeys (and whatever else should be convertible)